src/app/providers/colour-contrast/colour-contrast.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ColourContrastService {
+  private relativeLuminance(value: number): number {
+    const ratio = value / 255;
+    return ratio <= 0.04045 ? ratio / 12.92 : Math.pow((ratio + 0.055) / 1.055, 2.4);
+  }
+  private luminance(color: [number, number, number]): number {
```

Comment:
This can be destructed in line i.e.

`private luminance([colour1, colour2, colour3]: [number, number, number]): number {`